### PR TITLE
Migrate azure-storage-file-datalake to azure-xml

### PR DESCRIPTION
**Pull request by ibrandes**

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/LeaseAction.java [hunk 1]
```diff
@@ -4,24 +4,33 @@
...
 package com.azure.storage.file.datalake.models;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonValue;
-
-/** Defines values for LeaseAction. */
+/**
+ * Defines values for LeaseAction.
+ */
 public enum LeaseAction {
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/LeaseAction.java:10]: Why is there a `LeaseAction` in `implementation.models` and another in `models`? Can one of them be deleted?

File: sdk/storage/azure-storage-file-datalake/pom.xml [hunk 2]
```diff
@@ -78,11 +78,10 @@
...
       <version>12.27.0-beta.1</version> <!-- {x-version-update;com.azure:azure-storage-blob;current} -->
     </dependency>
-
     <dependency>
-      <groupId>com.fasterxml.jackson.dataformat</groupId>
-      <artifactId>jackson-dataformat-xml</artifactId>
-      <version>2.13.5</version> <!-- {x-version-update;com.fasterxml.jackson.dataformat:jackson-dataformat-xml;external_dependency} -->
+      <groupId>com.azure</groupId>
+      <artifactId>azure-json</artifactId>
+      <version>1.1.0</version> <!-- {x-version-update;com.azure:azure-json;dependency} -->
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/pom.xml:84]: This should be importing azure-xml and not azure-json

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/models/BlobHierarchyListSegment.java [hunk 3]
```diff
@@ -63,12 +67,105 @@ public List<BlobItemInternal> getBlobItems() {
...
+            }
+            xmlWriter.writeEndElement();
+        }
+        if (this.blobItems != null) {
+            xmlWriter.writeStartElement("BlobItems");
+            for (BlobItemInternal element : this.blobItems) {
+                xmlWriter.writeXml(element, "Blob");
+            }
+            xmlWriter.writeEndElement();
+        }
```
[the remark this answers is not in the document]
Comment (alzimmermsft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/models/BlobHierarchyListSegment.java:101]: You can follow this as an example:

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-file-share/swagger/src/main/java/ShareStorageCustomization.java#L37

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/models/Path.java [hunk 4]
```diff
@@ -318,12 +313,81 @@ public String getETag() {
...
+                reader.nextToken();
+
+                if ("name".equals(fieldName)) {
+                    deserializedPath.name = reader.getString();
+                } else if ("isDirectory".equals(fieldName)) {
+                    deserializedPath.isDirectory = reader.getNullable(JsonReader::getBoolean);
+                } else if ("lastModified".equals(fieldName)) {
+                    deserializedPath.lastModified = reader.getString();
+                } else if ("contentLength".equals(fieldName)) {
+                    deserializedPath.contentLength = reader.getNullable(JsonReader::getLong);
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/models/Path.java:368]: Appears the service returns this as a "String" so we'll need a transform here. I'd say to cover all bases it'd be good to do something more general like

```java
JsonToken token = reader.currentToken();
if (token == JsonToken.STRING) {
    deserializedPath.contentLength = Long.parseLong(reader.getString());
} else if (token == JsonToken.NUMBER) {
    deserializedPath.contentLength = reader.getLong();
} else if (token == JsonToken.NULL) {
    deserializedPath.contentLength = null;
} else {
    throw new IllegalStateException("Invalid token, expected one of STRING, NUMBER, or NULL. Was " + token);
}

**Review by alzimmermsft — APPROVED** [review]

**Review by ibrahimrabab — APPROVED** [review]

**Review by mssfang — APPROVED** [review]

**Pull request merged**